Makefile

```diff
@@ -1,25 +1,68 @@
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+OUTPUT_DIR := $(ROOT_DIR)/output
+BUILD_DIR := $(OUTPUT_DIR)/build
+IMAGE_DIR := $(OUTPUT_DIR)/images
+INITRAMFS_DIR := $(OUTPUT_DIR)/initramfs
 
-all: buildroot/Makefile buildroot/.config
-	make -C buildroot
+BR_INIT_ENV := BR2_EXTERNAL=$(ROOT_DIR)
 
+
+.PHONY: all
+all: $(IMAGE_DIR)/rootfs.img $(IMAGE_DIR)/uImage $(IMAGE_DIR)/am335x-kno_txt.dtb
+
+.PHONY: clean
 clean:
-	BR2_EXTERNAL=.. make -C buildroot clean
+	rm -rf $(OUTPUT_DIR)
+
+.PHONY: prepare-structure
+prepare-structure:
+	mkdir -p $(OUTPUT_DIR)
+	mkdir -p $(IMAGE_DIR)
+	mkdir -p $(INITRAMFS_DIR)
+
+.PHONY: prepare
+prepare: prepare-structure $(BUILD_DIR)/rootfs/.config $(BUILD_DIR)/initramfs/.config
+
+.PHONY: source
+source: rootfs-source initramfs-source
 
-buildroot/Makefile:
+.PHONY: rootfs-source
+rootfs-source: $(BUILD_DIR)/rootfs/.config
+	$(MAKE) -C $(BUILD_DIR)/rootfs source
+
+.PHONY: initramfs-source
+initramfs-source: $(BUILD_DIR)/initramfs/.config
+	$(MAKE) -C $(BUILD_DIR)/initramfs source
+
+$(BUILD_DIR)/rootfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/rootfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_rootfs_defconfig
+
+$(BUILD_DIR)/initramfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/initramfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_initramfs_defconfig
```

Comment:
Just to be clear: That means that the source packets are downloaded once and some packets like gcc are compiled multiple times, right? `make savedefconfig`  should work as before.